app/src/pages/public/PublicProjectPage.tsx

```diff
@@ -34,6 +35,8 @@ const PublicProjectPage = () => {
   const restorationTrackerApi = useRestorationTrackerApi();
   const classes = useStyles();
 
+  const codes = useCodes();
```

Comment:
Can be `const { codes } = useCodes();`, which prevents from writting codes.codes when passing it to children.

Reply:
For sure, but I am using `codes.isReady` as well as `codes.codes`.

`codes.codes` is a bit funny, maybe could update it to `codes.values` or something, but won't worry about that just now.

Reply:
(`codes.isReady` is a bit of a funny check, since it doesn't tell you if the codes are valid and we end up checking `codes.codes`, so I could potentially remove it.  I am also not fully utilizing `isLoading` inside the hook I just noticed.  I don't check it in the useEffect.  So there are some potential enhancements there.  But for another PR).